pandas/core/arrays/_mixins.py

```diff
@@ -276,10 +277,20 @@ def __setitem__(self, key, value):
     def _validate_setitem_value(self, value):
         return value
 
+    @overload
+    def __getitem__(self, item: int | np.integer) -> Any:
+        ...
+
+    @overload
     def __getitem__(
-        self: NDArrayBackedExtensionArrayT,
+        self: NDArrayBackedExtensionArray, item: slice | np.ndarray | Sequence[int]
+    ) -> NDArrayBackedExtensionArray:
```

Comment:
is there a TypeVar for this?

Reply:
> is there a TypeVar for this?

Yes, there is, but there is no need to use it.

Reply:
please do this for consistency

Reply:
done in next commit

Reply:
did the commit using the TypeVar not get pushed?

Reply:
NDArrayBackedExtensionArrayT?

Reply:
fixed in next commit.  Not sure how I missed that.